frankenphp.go

```diff
@@ -69,6 +70,8 @@ var (
 
 	loggerMu sync.RWMutex
 	logger   *zap.Logger
+
+	metrics Metrics = NullMetrics{}
```

Comment:
Can't we use `nil` instead to prevent adding/maintaining a dedicated symbol?

Reply:
I did that at first, but I found the constant typing of `if metrics != nil` to be harder to maintain. In this way, we don't have any branching and the compiler can compile out the entire metrics system in the event a caller never sets metrics up.

Reply:
I think I prefer the branches, but why not.
Could we at least make this symbol internal? (`nullMetrics`)?